src/check-static-inits.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+import sys, os, os.path, shutil, subprocess, glob, re
+
+builddir = os.environ.get ('builddir', os.path.dirname (__file__))
+libs = os.environ.get ('libs', '.libs')
+
+objdump = shutil.which ('objdump')
+if not objdump:
+	print ('check-static-inits.py: \'ldd\' not found; skipping test')
+	sys.exit (77)
+
+if sys.version_info < (3, 5):
+	print ('check-static-inits.py: needs python 3.5 for recursive support in glob')
+	sys.exit (77)
+
+OBJS = glob.glob (os.path.join (builddir, libs, '**', '*.o'), recursive=True)
+if not OBJS:
+	print ("check-static-inits.py: object files not found; skipping test")
+	sys.exit (77)
+
+stat = 0
+
+for obj in OBJS:
+	result = subprocess.check_output ([objdump, '-t', obj]).decode ('utf-8')
+
+	# Checking that no object file has static initializers
+	for l in re.findall (r'^.*\.[cd]tors.*$', result, re.MULTILINE):
```

Comment:
This matches https://github.com/harfbuzz/harfbuzz/blob/0d8442a55f6b34ca3b5f444de2a3cdcce27421ff/src/check-static-inits.sh#L27 logic AFAIK but I couldn't reproduce and test the check using `__attribute__((constructor))` or C++ static initialized variables, something tried more or less for other tests.

Reply:
Did you try a static initialized variable in a method?

These tests were developed according to what common compilers were producing at the time.  It's possible some of them are not relevant anymore.  But this one should be still relevant...

Reply:
> C++ static initialized variables

With a non-trivial constructor?

I made the following trivial patch to test, but am not on Linux and don't have `objdump`, so this test is skipping for me.
```patch
+static hb_set_t oh_oh;
 /**
  * hb_set_previous_range:
  * @set: a set.
@@ -537,5 +538,6 @@ hb_set_previous_range (const hb_set_t *set,
                       hb_codepoint_t *first,
                       hb_codepoint_t *last)
 {
+  if (oh_oh.has (*first)) return true;
   return set->previous_range (first, last);
 }
```

Reply:
The example triggers the next test and exposes `__cxa_atexit` but not `.ctors` and `.dtors`

> It's possible some of them are not relevant anymore.

Sure! Completely understandable and nice to have tests, thank you very much! :)

Reply:
> Sure! Completely understandable and nice to have tests, thank you very much! :)

IIRC Chrome was porting to harfbuzz-ng then and they brought this to my attention.  It's possible that the first test is not what I could reproduce and I added the second one to work on whatever setup I had.

I just tested both gcc and clang on Fedora and neither emit those segments in anything I tried.  I can dig harder to find, but as you said, doesn't matter. :)